src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,151 @@
+                            player, "" + noteBlock.get(INSTRUMENT) + "[" + noteBlock.get(NOTE) + "]"
+                        )
+                        map.add(noteBlock.get(INSTRUMENT), noteBlock.get(NOTE), BlockPos(pos))
+                    }
+                }
+            }
+            NotebotMode.LOAD -> {
+                noteSequence = MidiParser.parse("songs" + File.separator + songName, 5)
+                if (noteSequence.isEmpty())
+                    snackbarMessage(player, "Song not found: " + "songs" + File.separator + songName)
+                else
+                    snackbarMessage(player, "Song Found: " + "songs" + File.separator + songName)
+                lastNote = System.currentTimeMillis()
+                elapsed = 0
+            }
+            NotebotMode.PLAY -> {
+                if (!player.isCreative) {
+                    while (noteSequence.isNotEmpty() && noteSequence.firstKey() <= elapsed) {
+                        playNotes(noteSequence.pollFirstEntry().value, player, world)
+                    }
+                    elapsed += System.currentTimeMillis() - lastNote
+                    lastNote = System.currentTimeMillis()
+                } else snackbarMessage(player, "You are in creative mode and cannot play music.")
+            }
+        }
+    })
+
+    private fun playNotes(notes: ArrayList<Note>, player: ClientPlayerEntity, world: ClientWorld) {
+        val blockPosArr: ArrayList<BlockPos?> = ArrayList()
+        notes.forEach { n ->
+            channelsArray = arrayOf(ChannelZero, ChannelOne, ChannelTwo, ChannelThree, ChannelFour)
+            val number = (n.track % (channelsArray.size - 1))
+            val enum = channelsArray[if (number < 0) 0 else number]
+            snackbarMessage(player, enum.toString())
+            blockPosArr.add(map[enum][if (enum == Instrument.HAT && hatAlwaysAsFSharp) 0 else n.notebotNote])
+        }
+        playBlock(blockPosArr, player, world)
+    }
+
+    fun snackbarMessage(player: ClientPlayerEntity, t: String) {
+        player.sendMessage(text { +t }, true)
+    }
+
+    private fun playBlock(blockPosTracks: ArrayList<BlockPos?>, player: ClientPlayerEntity, world: ClientWorld) {
+        if (blockPosTracks.isNotEmpty()) {
+            blockPosTracks.forEach { blockPos ->
+                if (blockPos != null) {
+                    getIrreplaceableNeighbour(world, blockPos)?.let { (_, direction) ->
+                        mc.networkHandler?.sendPacket(
+                            PlayerActionC2SPacket(PlayerActionC2SPacket.Action.START_DESTROY_BLOCK, blockPos, direction)
+                        )
+                        mc.networkHandler?.sendPacket(
+                            PlayerActionC2SPacket(PlayerActionC2SPacket.Action.ABORT_DESTROY_BLOCK, blockPos, direction)
+                        )
+                        player.swingHand(Hand.MAIN_HAND)
+                        (mc as IMinecraftClient).setItemUseCooldown(4)
```

Comment:
This is pretty inconsistent throughout the codebase, but use an extension property to hide the ugly cast. Look in `Extensions.kt` for how to make one, and then just call it like this:
```suggestion
                        mc.itemUseCooldown = 4
```